gitlab/integration_test.go

```diff
@@ -886,6 +886,100 @@ var _ = Describe("GitLab Provider", func() {
 
 	})
 
+	It("should be possible to download files from path and branch specified with nested directory", func() {
+
+		userRepoRef := newUserRepoRef(testUserName, testRepoName)
+
+		userRepo, err := c.UserRepositories().Get(ctx, userRepoRef)
+		Expect(err).ToNot(HaveOccurred())
+
+		defaultBranch := userRepo.Get().DefaultBranch
+
+		path0 := "clustersDir/cluster/machine.yaml"
+		content0 := "machine0 yaml content"
+		path1 := "clustersDir/cluster/machine1.yaml"
+		content1 := "machine1 yaml content"
+		path2 := "clustersDir/cluster2/clusterSubDir/machine2.yaml"
+		content2 := "machine2 yaml content"
+
+		files := []gitprovider.CommitFile{
+			{
+				Path:    &path0,
+				Content: &content0,
+			},
+			{
+				Path:    &path1,
+				Content: &content1,
+			},
+			{
+				Path:    &path2,
+				Content: &content2,
+			},
+		}
+
+		commitFiles := make([]gitprovider.CommitFile, 0)
+		for _, file := range files {
+			path := file.Path
+			content := file.Content
+			commitFiles = append(commitFiles, gitprovider.CommitFile{
+				Path:    path,
+				Content: content,
+			})
+		}
+
+		_, err = userRepo.Commits().Create(ctx, *defaultBranch, "added files", commitFiles)
+		Expect(err).ToNot(HaveOccurred())
+
+		options := gitprovider.FilesGetOptions{Recursive: *gitprovider.BoolVar(true)}
+		downloadedFiles, err := userRepo.Files().Get(ctx, "clustersDir", *defaultBranch, &options)
+		Expect(err).ToNot(HaveOccurred())
+		for ind, downloadedFile := range downloadedFiles {
+			Expect(*downloadedFile).To(Equal(files[ind]))
+		}
+
+	})
+	It("should be possible list repo tree files", func() {
+		userRepoRef := newUserRepoRef(testUserName, testRepoName)
+
+		userRepo, err := c.UserRepositories().Get(ctx, userRepoRef)
+		Expect(err).ToNot(HaveOccurred())
+
+		defaultBranch := userRepo.Get().DefaultBranch
+
+		path0 := "clustersDir/cluster/machine.yaml"
+		content0 := "machine0 yaml content"
+		path1 := "clustersDir/cluster/machine1.yaml"
+		content1 := "machine1 yaml content"
+		path2 := "clustersDir/cluster2/clusterSubDir/machine2.yaml"
+		content2 := "machine2 yaml content"
+
+		files := []gitprovider.CommitFile{
+			{
+				Path:    &path0,
+				Content: &content0,
+			},
+			{
+				Path:    &path1,
+				Content: &content1,
+			},
+			{
+				Path:    &path2,
+				Content: &content2,
+			},
+		}
```

Comment:
Should these files be committed to the repo as in the above test?

Maybe we could add 1 more file not in clusterDir just to check the path filtering?

Reply:
the commit from the previous test includes these files, and so listed the files from the same commit created earlier.
A file from outside the path given can be added yes

Reply:
@foot actually there is README.md file in the repo that is not included in the files retrieved as it is outside the specified directory

Reply:
💯 !